src/python_inspector/utils_pypi.py

```diff
@@ -1593,8 +1593,15 @@ def fetch_links(
         for anchor_tag in anchor_tags:
             python_requires = None
             url, _, _sha256 = anchor_tag["href"].partition("#sha256=")
+            if url.startswith(".."):
+                # Handle relative links
+                url = urljoin(package_url, url)
             if "data-requires-python" in anchor_tag.attrs:
                 python_requires = anchor_tag.attrs["data-requires-python"]
+            # Check if the link is a relative URL
+            if not url.startswith(("http://", "https://")):
+                base_url = "/".join(package_url.split("/")[:-1])  # Extract base URL
```

Comment:
Please extract this part of code in a separate function and add some doctest or unit tests for same.

Reply:
He llo @TG1999 , i have already added 2 new files for the test
tests/data/fetch_links_test.html
tests/data/relative-links-expected.json
and modified 
tests/test_utils.py

Those tests aren't suitable?

Reply:
@TG1999 Could you please provide a feedback, is the change ready for merge?